tests/python/utils.py

```diff
@@ -242,13 +242,24 @@ def check_captured_python_definition(reference_outputs, fd, inputs, device=None)
 
 # Run original FusionDefinition
 # Clone FusionDefinition
+# Apply segmentation if it supported for this FusionDefinition
 # Run cloned python definition
 # Check that the result of cloned python definition matches original results
-def check_cpp_translation(reference_outputs, fd, inputs, device=None):
+def check_cpp_translation(
+    reference_outputs, fd, inputs, supports_segmentation, device=None
+):
     try:
         torch.manual_seed(0)
+
+        # Clone
         cloned_fd = FusionDefinition()
         clone(fd, cloned_fd)
+
+        # Segment
+        if supports_segmentation:
+            cloned_fd.segment(inputs)
```

Comment:
what's the expected workflow in the finished form?

Looks like `segment` should always be called and check if segmentation happens? hint: maybe we want to name it as `_segment` so it's not intended to be exposed as a user API?

Reply:
I'm wondering if we should also consider having `supports_segmentation` as an optional kwargs in `FusionDefinition.execute`

Reply:
> Looks like segment should always be called and check if segmentation happens? hint: maybe we want to name it as _segment so it's not intended to be exposed as a user API?

btw, I'm not saying that we should refactor that in this PR.

I understand that the sole purpose of segment usage is for debugging. I think we can simplify that and have it done automatically. We can discuss this later maybe.

Reply:
```python
with FusionDefinition() as fd:
  # define fusion
fd.segment(inputs)  # manually segment fusion
fd.execute(inputs) # run fusion
```
This ^^^ is the current workflow. The user calls `segment` and runs `execute`.

Reply:
Your suggestion is:
```python
with FusionDefinition() as fd:
  # define fusion
fd.execute(inputs, segment_fusion=True) # run `fusion` and `segment` in `execute`
```

Reply:
Sorta.

My question is, when would we expect someone to use `segment_fusion=True`? If the expectation is just for getting a smaller repro, we can automate this. i.e. user don't need to enable python segmentation when they execute, we can do that in the try/catch block and minimize the repro there.

But if we plan to have people eagerly segment and play with scheduling for each segments, that's a different story.

Reply:
It is the latter for me. I'd like to replace badly segmented fusions with a manually crafted solution. Other ideas is to prototype new segmentation algorithms in python.